### PR TITLE
Move to default_instance QSPIF

### PR DESCRIPTION
**Pull request by manchoz**

### CODE REVIEW

**Comment by manchoz** [comment]
@pennam yes, it is!

It is needed for all the scenarios where OTA needs to be done via media other than WiFi (USB, Ethernet, etc.).

When `WiFi.h` is not included the QSPIF is not initialized and the `begin` method and its submethods will fail.

**Comment by manchoz** [comment]
@pennam please, review.

File: src/Arduino_Portenta_OTA_QSPI.cpp [hunk 1]
```diff
@@ -47,13 +43,7 @@ Arduino_Portenta_OTA_QSPI::Arduino_Portenta_OTA_QSPI(StorageTypePortenta const s
 
 bool Arduino_Portenta_OTA_QSPI::init()
 {
-#if !defined (COMPONENT_4343W_FS)
-  qspi_bd = new QSPIFBlockDevice(PD_11, PD_12, PF_7, PD_13,  PF_10, PG_6, QSPIF_POLARITY_MODE_1, 40000000);
```
Comment (pennam) [inline src/Arduino_Portenta_OTA_QSPI.cpp:51]: If we remove completely this line we are dropping compatibility for older core versions  ( < 2.3.1 )

File: src/Arduino_Portenta_OTA_QSPI.cpp [hunk 2]
```diff
@@ -47,13 +43,7 @@ Arduino_Portenta_OTA_QSPI::Arduino_Portenta_OTA_QSPI(StorageTypePortenta const s
...
-  if (qspi_bd->init() != QSPIF_BD_ERROR_OK) {
-    Serial1.println("Error: QSPI init failure.");
-    return false;
-  }
-#endif
+  auto qspi_bd = mbed::BlockDevice::get_default_instance();
```
Comment (pennam) [inline src/Arduino_Portenta_OTA_QSPI.cpp:46]: We still need to call the init function here, even if we are using `get_default_instance()`

**Review by pennam — COMMENTED** [review]
@manchoz could you please have a look to this commit https://github.com/pennam/Arduino_Portenta_OTA/commit/7f37a53cb95aabb5c80d77ebe31def9bda32b6d8 ? Doing like this we are keeping compatibility with older core versions and using `get_default_instance() `for the new ones. I've also added a call to the `init()` method.

**Comment by pennam** [comment]
superseded by #21

**Pull request closed without merging**